api/api/V1/AuthenticateUser.ts

```diff
@@ -223,4 +224,58 @@ export default function (app: Application) {
       });
     })
   );
+
+  app.post(
+    "/resetpassword",
+    validateFields([
+      oneOf(
+        [
+          deprecatedField(validNameOf(body("username"))),
+          validNameOf(body("userName")),
+          validNameOf(body("nameOrEmail")),
+          body("nameOrEmail").isEmail(),
+          body("email").isEmail(),
+        ],
+        "Missing user name in request"
+      ),
+    ]),
+    fetchUnauthorizedOptionalUserByNameOrEmailOrId(
```

Comment:
Shouldn't this be `fetchUnauthorizedRequiredUserByNameOrEmailOrId`?  Then no need to check for the user later.

Reply:
This is a security thing so no matter what you never know if you have gotten an invalid or valid user, when you try to reset a password. i.e. always want to return 200. I think this is normal?